src/sub/tests/unit/test_subhub.py

```diff
@@ -366,3 +387,420 @@ def test_plan_response_invalid(mock_plans, app):
+        assert response_body == expected_response_body
+
+    def test_different_intervals(self):
+        self.get_subhub_user_mock.return_value = self.subhub_user
+        self.retrieve_customer_mock.return_value = self.customer
+        self.retrieve_plan_mock.return_value = self.plan2
+
+        expected_response_body = {
+            "errno": 1002,
+            "message": "The plans do not have the same interval",
+        }
+        data = {"plan_id": "plan_test2"}
+
+        response = self.client.patch(
+            "v1/sub/customer/user123/subscriptions/sub_test1",
+            headers={"Authorization": "fake_payment_api_key"},
+            data=json.dumps(data),
+            content_type="application/json",
+        )
+
+        response_body = json.loads(response.data)
+        assert response.status_code == 400
+        assert response_body == expected_response_body
+
+    def test_different_intervals_count(self):
+        self.get_subhub_user_mock.return_value = self.subhub_user
+        self.retrieve_customer_mock.return_value = self.customer
+        self.retrieve_plan_mock.return_value = self.plan5
+
+        expected_response_body = {
+            "errno": 1002,
+            "message": "The plans do not have the same interval",
+        }
+        data = {"plan_id": "plan_test2"}
+
+        response = self.client.patch(
+            "v1/sub/customer/user123/subscriptions/sub_test1",
+            headers={"Authorization": "fake_payment_api_key"},
+            data=json.dumps(data),
+            content_type="application/json",
+        )
+
+        response_body = json.loads(response.data)
+        assert response.status_code == 400
+        assert response_body == expected_response_body
+
+    def test_plan_not_found(self):
+        self.get_subhub_user_mock.return_value = self.subhub_user
+        self.retrieve_customer_mock.return_value = self.customer
+        self.retrieve_plan_mock.side_effect = stripe.error.InvalidRequestError(
+            "message", param="param", http_status=404
+        )
+
+        expected_response_body = {"errno": 4003, "message": "Plan not found"}
+        data = {"plan_id": "plan_test2"}
+
+        response = self.client.patch(
+            "v1/sub/customer/user123/subscriptions/sub_test1",
+            headers={"Authorization": "fake_payment_api_key"},
+            data=json.dumps(data),
+            content_type="application/json",
+        )
+
+        response_body = json.loads(response.data)
+        assert response.status_code == 404
+        assert response_body == expected_response_body
+
+    def test_plan_fetch_error(self):
+        self.get_subhub_user_mock.return_value = self.subhub_user
+        self.retrieve_customer_mock.return_value = self.customer
+        self.retrieve_plan_mock.side_effect = stripe.error.InvalidRequestError(
+            "message", param="param"
+        )
+
+        data = {"plan_id": "plan_test2"}
+
+        response = self.client.patch(
```

Comment:
this block is repeated over and over, could be a good candidate for a function that you just pass the `data` to because it looks like everything else is the same every time

Reply:
Unique tests.